### ISSUE TITLE
ES6 named exports are caught by no-unused-vars

### ISSUE THREAD

**Issue by hawkrives** [eslint/eslint#2149]
```js
let toUpper = (partial) => partial.toUpperCase
export {toUpper}
```

fails with

```
<text>
  1:9  warning  toUpper is defined but never used              no-unused-vars
```

I would expect that `toUpper` would be seen as being used by the named export.

**Comment by nzakas** [comment]
Yeah, that's a bug.

**Comment by nzakas** [comment]
Working on this.

**Comment by nzakas** [comment]
This one is too tough to fix without updating escope, so we'll have to wait for that.

**Comment by hawkrives** [comment]
Sounds good. Thanks for looking into this.

**Comment by nzakas** [comment]
Fixed by #2235
